zerver/lib/send_email.py

```diff
@@ -149,8 +150,15 @@ def render_templates() -> Tuple[str, str, str]:
     if from_address == FromAddress.support_placeholder:
         from_address = FromAddress.SUPPORT
 
-    # Set the "From" that is displayed separately from the envelope-from
-    extra_headers["From"] = str(Address(display_name=from_name, addr_spec=from_address))
+    # Set the "From" that is displayed separately from the envelope-from.
+    # Check that the sender address is not overly-long for SES compatibility.
+    # If the address is still too long without the name an SMTPDataError will
+    # be raised when sending the email and it will be caught to raise an
+    # EmailNotDeliveredException
+    if getsizeof(from_name + from_address) > 320:
+        extra_headers["From"] = str(Address(addr_spec=from_address))
+    else:
+        extra_headers["From"] = str(Address(display_name=from_name, addr_spec=from_address))
```

Comment:
This looks right to me, but we should have a test for this.

`git grep outbox zerver/tests` will find you basically every test that sends outgoing emails and you can hopefully find a good existing test to crib from.  (I would just try to find one that's nice and simple, and then give a user the appropriate name and verify things).

Reply:
@alexmv do you recall the setup for what code path was sending an email?  I'm not entirely sure how one gets to 320 characters starting with a 100-character limit for the `full_name` field, but I guess it's encoding UTF-8 as several characters per word?

Reply:
#17558 shows what we saw from production -- an email address of the form:
```
From: ȩ̷̡̢̡͍͙̠̹̠̮̭͓͇͈̝̤̟͂̽͒̊͌͒̽͋̈͆͋͂͜͝ͅẍ̸̨̧̻̞̪̞̹̳̅͌̾͊͒̂͊́̇̈́͑͒́͛͆̏̋̚͘͜͝͠a̷̟̠̹̪̣͊̈̽̌͑̈́̋̍̀̌̽̾̋͊͊̇̓̌̍̒̕̚͠͝͝m̴̢̢̨̢̙̰̙̺̠̭͎̮͓͇̯̗̻̮͊̔̐̕͝ͅͅp̷̨̢̛͈͙̫̫̞̤͉̭̯̯̖̝̹̝̝̱̯̖̠̹͔̅̓͊̓͑̈͌͌̾̾̚͘͘͜l̸̢̹̰̦̪͕̟̲̰̦̫̳͔̯̻̂́̏̋̌̀̆̐̂́̐̄̏͒͐̈̈̈͘͜ẽ̸̛̜̜͐͆̓͛̃̉̈͛̓̿̅̅̑̕ <example@example.com>
```